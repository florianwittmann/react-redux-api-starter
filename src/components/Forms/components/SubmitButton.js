import React from "react";

export default props => {
  return (
    <div className="field">
      <div className="control">

        <button
          type="submit"
          className={`button is-primary ${props.inProgress ? "is-loading" : ""}`}
        >
          {props.children}
        </button>
      </div>
    </div>
  );
};
