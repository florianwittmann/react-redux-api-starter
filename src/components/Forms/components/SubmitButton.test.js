import React from "react";
import { shallow } from "enzyme";
import SubmitButton from "./SubmitButton";

test("should render SubmitButton", () => {
  const wrapper = shallow(<SubmitButton>Some Button</SubmitButton>);
  expect(wrapper).toMatchSnapshot();
});

test("should render SubmitButton inProgress", () => {
  const wrapper = shallow(
    <SubmitButton inProgress={true}>Some Button</SubmitButton>
  );
  expect(wrapper).toMatchSnapshot();
});
