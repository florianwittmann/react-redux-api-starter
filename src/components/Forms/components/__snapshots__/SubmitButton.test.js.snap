// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SubmitButton 1`] = `
<div
  className="field"
>
  <div
    className="control"
  >
    <button
      className="button is-primary "
      type="submit"
    >
      Some Button
    </button>
  </div>
</div>
`;

exports[`should render SubmitButton inProgress 1`] = `
<div
  className="field"
>
  <div
    className="control"
  >
    <button
      className="button is-primary is-loading"
      type="submit"
    >
      Some Button
    </button>
  </div>
</div>
`;
